src/test/blockencodings_tests.cpp

```diff
@@ -302,6 +303,44 @@ BOOST_AUTO_TEST_CASE(EmptyBlockRoundTripTest)
     }
 }
 
+BOOST_AUTO_TEST_CASE(ReceiveWithExtraTransactions) {
+    CTxMemPool& pool = *Assert(m_node.mempool);
+    TestMemPoolEntryHelper entry;
+    CBlock block(BuildBlockTestCase());
+    std::vector<CTransactionRef> extra_txn;
+    extra_txn.resize(10);
+
+    CMutableTransaction mtx = BuildTransactionTestCase();
+    mtx.vin[0].prevout.hash = Txid::FromUint256(InsecureRand256());
+    mtx.vin[0].prevout.n = 0;
+    CTransactionRef non_block_tx = MakeTransactionRef(std::move(mtx));
+
+    LOCK2(cs_main, pool.cs);
+    pool.addUnchecked(entry.FromTx(block.vtx[2]));
+    BOOST_CHECK_EQUAL(pool.get(block.vtx[2]->GetHash()).use_count(), SHARED_TX_OFFSET + 0);
+
+    {
+        CBlockHeaderAndShortTxIDs cmpctblock{block};
+        PartiallyDownloadedBlock partialBlock(&pool);
+        PartiallyDownloadedBlock partialBlockWithExtra(&pool);
```

Comment:
nit: We prefer using snake case for new code, see https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c.